File: backend-go/internal/modules/system/controller/job_controller.go
package controller

import (
	"drip-admin/backend-go/internal/common"
	"drip-admin/backend-go/internal/modules/system/dto"
	"github.com/gin-gonic/gin"
)

var (
	_ = common.ApiResponse{}
	_ = dto.JobSaveRequest{}
)

// Jobs returns scheduled jobs.
// @Summary List jobs
// @Tags job
// @Produce json
// @Security ApiKeyAuth
// @Param page query int false "Page"
// @Param pageSize query int false "Page size"
// @Param jobName query string false "Job name"
// @Param remark query string false "Remark"
// @Param status query int false "Status"
// @Param createdAt query string false "Created time"
// @Success 200 {object} common.ApiResponse
// @Failure 401 {object} common.ApiResponse
// @Failure 403 {object} common.ApiResponse
// @Router /system/job [get]
func (ctl *Controller) Jobs(c *gin.Context) {
	ctl.svc.Jobs(c)
}

// Job returns a scheduled job detail.
// @Summary Job detail
// @Tags job
// @Produce json
// @Security ApiKeyAuth
// @Param id path string true "Job ID"
// @Success 200 {object} common.ApiResponse
// @Failure 401 {object} common.ApiResponse
// @Failure 403 {object} common.ApiResponse
// @Failure 404 {object} common.ApiResponse
// @Router /system/job/{id} [get]
func (ctl *Controller) Job(c *gin.Context) {
	ctl.svc.Job(c)
}

// JobScripts returns available job scripts.
// @Summary Job scripts
// @Tags job
// @Produce json
// @Security ApiKeyAuth
// @Param executorType query string false "Executor type: python, shell, bat, powershell, ps1"
// @Success 200 {object} common.ApiResponse
// @Failure 401 {object} common.ApiResponse
// @Failure 403 {object} common.ApiResponse
// @Router /system/job/scripts [get]
func (ctl *Controller) JobScripts(c *gin.Context) {
	ctl.svc.JobScripts(c)
}

// CreateJob creates a scheduled job.
// @Summary Create job
// @Tags job
// @Accept json
// @Produce json
// @Security ApiKeyAuth
// @Param request body dto.JobSaveRequest true "Request"
// @Success 200 {object} common.ApiResponse
// @Failure 400 {object} common.ApiResponse
// @Failure 401 {object} common.ApiResponse
// @Failure 403 {object} common.ApiResponse
// @Router /system/job [post]
func (ctl *Controller) CreateJob(c *gin.Context) {
	ctl.svc.CreateJob(c)
}

// UpdateJob updates a scheduled job.
// @Summary Update job
// @Tags job
// @Accept json
// @Produce json
// @Security ApiKeyAuth
// @Param id path string true "Job ID"
// @Param request body dto.JobSaveRequest true "Request"
// @Success 200 {object} common.ApiResponse
// @Failure 400 {object} common.ApiResponse
// @Failure 401 {object} common.ApiResponse
// @Failure 403 {object} common.ApiResponse
// @Failure 404 {object} common.ApiResponse
// @Router /system/job/{id} [put]
func (ctl *Controller) UpdateJob(c *gin.Context) {
	ctl.svc.UpdateJob(c)
}

// DeleteJob deletes a scheduled job.
// @Summary Delete job
// @Tags job
// @Produce json
// @Security ApiKeyAuth
// @Param id path string true "Job ID"
// @Success 200 {object} common.ApiResponse
// @Failure 401 {object} common.ApiResponse
// @Failure 403 {object} common.ApiResponse
// @Failure 404 {object} common.ApiResponse
// @Router /system/job/{id} [delete]
func (ctl *Controller) DeleteJob(c *gin.Context) {
	ctl.svc.DeleteJob(c)
}

// JobStatus updates a scheduled job status.
// @Summary Update job status
// @Tags job
// @Accept json
// @Produce json
// @Security ApiKeyAuth
// @Param id path string true "Job ID"
// @Param request body dto.StatusUpdateRequest true "Request"
// @Success 200 {object} common.ApiResponse
// @Failure 400 {object} common.ApiResponse
// @Failure 401 {object} common.ApiResponse
// @Failure 403 {object} common.ApiResponse
// @Router /system/job/{id}/status [put]
func (ctl *Controller) JobStatus(c *gin.Context) {
	ctl.svc.JobStatus(c)
}

// RunJob runs a scheduled job once.
// @Summary Run job
// @Tags job
// @Produce json
// @Security ApiKeyAuth
// @Param id path string true "Job ID"
// @Success 200 {object} common.ApiResponse
// @Failure 401 {object} common.ApiResponse
// @Failure 403 {object} common.ApiResponse
// @Failure 404 {object} common.ApiResponse
// @Router /system/job/{id}/run [post]
func (ctl *Controller) RunJob(c *gin.Context) {
	ctl.svc.RunJob(c)
}

// JobLogs returns run logs for a scheduled job.
// @Summary Job run logs
// @Tags job
// @Produce json
// @Security ApiKeyAuth
// @Param id path string true "Job ID"
// @Param page query int false "Page"
// @Param pageSize query int false "Page size"
// @Success 200 {object} common.ApiResponse
// @Failure 401 {object} common.ApiResponse
// @Failure 403 {object} common.ApiResponse
// @Router /system/job/{id}/runLog [get]
func (ctl *Controller) JobLogs(c *gin.Context) {
	ctl.svc.JobLogs(c)
}

// JobRunLogs returns all job run logs.
// @Summary List job run logs
// @Tags job
// @Produce json
// @Security ApiKeyAuth
// @Param page query int false "Page"
// @Param pageSize query int false "Page size"
// @Param jobName query string false "Job name"
// @Param status query string false "Status"
// @Success 200 {object} common.ApiResponse
// @Failure 401 {object} common.ApiResponse
// @Failure 403 {object} common.ApiResponse
// @Router /system/jobRunLog [get]
func (ctl *Controller) JobRunLogs(c *gin.Context) {
	ctl.svc.JobRunLogs(c)
}

File: backend-python/app/modules/system/controller/job_controller.py
from fastapi import APIRouter, Depends, Query, Request
from sqlalchemy.ext.asyncio import AsyncSession

from app.common.api_response import ApiResponse, success
from app.common.pagination import PageQuery
from app.modules.system.dto.job_request import JobSaveRequest
from app.modules.system.dto.user_request import StatusUpdateRequest
from app.modules.system.router import get_db, require_permission
from app.modules.system.service.job_service import JobService

router = APIRouter(tags=["job"])


def service(request: Request, db: AsyncSession = Depends(get_db)) -> JobService:
    return JobService(db, request.app.state.settings.job.script_dir)


@router.get("/job", response_model=ApiResponse)
async def jobs(
    page: PageQuery = Depends(),
    job_name: str | None = Query(default=None, alias="jobName"),
    remark: str | None = None,
    status: int | None = None,
    created_at: str | None = Query(default=None, alias="createdAt"),
    _: dict = Depends(require_permission("system:job:list")),
    current: JobService = Depends(service),
) -> ApiResponse:
    return success(await current.list_jobs(page, job_name, remark, status, created_at))


@router.get("/job/scripts", response_model=ApiResponse)
async def job_scripts(
    executor_type: str = Query(alias="executorType"),
    _: dict = Depends(require_permission("system:job:list")),
    current: JobService = Depends(service),
) -> ApiResponse:
    return success(current.executor.list_scripts(executor_type))


@router.get("/job/{id}", response_model=ApiResponse)
async def job_detail(
    id: int,
    _: dict = Depends(require_permission("system:job:list")),
    current: JobService = Depends(service),
) -> ApiResponse:
    return success(await current.detail(id))


@router.post("/job", response_model=ApiResponse)
async def create_job(
    body: JobSaveRequest,
    _: dict = Depends(require_permission("system:job:create")),
    current: JobService = Depends(service),
) -> ApiResponse:
    return success(await current.save(None, body))


@router.put("/job/{id}", response_model=ApiResponse)
async def update_job(
    id: int,
    body: JobSaveRequest,
    _: dict = Depends(require_permission("system:job:update")),
    current: JobService = Depends(service),
) -> ApiResponse:
    await current.save(id, body)
    return success()


@router.delete("/job/{id}", response_model=ApiResponse)
async def delete_job(
    id: int,
    _: dict = Depends(require_permission("system:job:delete")),
    current: JobService = Depends(service),
) -> ApiResponse:
    await current.delete(id)
    return success()


@router.put("/job/{id}/status", response_model=ApiResponse)
async def job_status(
    id: int,
    body: StatusUpdateRequest,
    _: dict = Depends(require_permission("system:job:update")),
    current: JobService = Depends(service),
) -> ApiResponse:
    await current.update_status(id, body.status)
    return success()


@router.post("/job/{id}/run", response_model=ApiResponse)
async def run_job(
    id: int,
    _: dict = Depends(require_permission("system:job:run")),
    current: JobService = Depends(service),
) -> ApiResponse:
    await current.run(id)
    return success()


@router.get("/job/{id}/runLog", response_model=ApiResponse)
async def job_run_logs(
    id: int,
    page: PageQuery = Depends(),
    job_name: str | None = Query(default=None, alias="jobName"),
    status: str | None = None,
    started_range: list[str] | None = Query(default=None, alias="startedRange"),
    _: dict = Depends(require_permission("system:job:list")),
    current: JobService = Depends(service),
) -> ApiResponse:
    return success(await current.run_logs(page, id, job_name, status, started_range))


@router.get("/jobRunLog", response_model=ApiResponse)
async def job_run_log_list(
    page: PageQuery = Depends(),
    job_name: str | None = Query(default=None, alias="jobName"),
    status: str | None = None,
    started_range: list[str] | None = Query(default=None, alias="startedRange"),
    _: dict = Depends(require_permission("system:job:history")),
    current: JobService = Depends(service),
) -> ApiResponse:
    return success(await current.run_logs(page, None, job_name, status, started_range))
